### PR TITLE
Add new bindings to ESS layer

### PR DESCRIPTION
**Pull request by andrew-christianson**
Reopening #4166 with on a separate branch in my fork + resolve the merge conflict with #4150.

Commit Message:

Add new evil-leader bindings for lots of the functions behind
ess-dev-map and ess-doc-map. Major additions are SPC m d for IDE-like
commands (roxygen, style), SPC m b for debugger interaction, and a few
extras in SPC m s and SPC m h.

Also add a few new layer variables controlling the indenting style, and
update README for all.

### CODE REVIEW

File: layers/+lang/ess/README.org [hunk 1]
```diff
@@ -69,3 +39,133 @@ hook:
...
+| ~SPC m s :~ | execute a command in the REPL, displaying the results in a separate buffer       |
+| ~SPC m v d~ | display a dired-like buffer browsing the objects in the inferior session buffer. |
+| ~CTRL+j~    | next item in REPL history                                                        |
+| ~CTRL+k~    | previous item in REPL history                                                    |
+
+** Developer
+Useful, IDE like functions.
+| Key Binding | Description                                                                                |
+|-------------+--------------------------------------------------------------------------------------------|
+| ~SPC m d l~ | ~(ess-developer-load-package)~ load a specified directory with ~devtools::load_all~        |
```
Comment (d12frosted) [inline layers/+lang/ess/README.org]: Please, avoid function names here. People usually far more interested in what it does 😸

**Comment by d12frosted** [comment]
Wow, this is an old one PR. Is it still relevant? 😸 

P. S. Sorry for taking so long 😢

**Comment by andrew-christianson** [comment]
Spacemacs + ESS is still my daily driver @ work so it's still relevant to me :).  I've been using the stock layer for a while now for convenience sake, though I'd love to come back to this and incorporate the above + resolve the conflicts if you guys would consider it.

File: layers/+lang/ess/README.org [hunk 2]
```diff
@@ -3,57 +3,63 @@
...
- - [[#options][Options]]
-   - [[#style][Style]]
-   - [[#roxygen][Roxygen]]
- - [[#key-bindings][Key Bindings]]
-   - [[#inferior-ess-processes][Inferior ESS Processes]]
-   - [[#evaluation-commands][Evaluation Commands]]
-   - [[#Developer][Developer]]
-   - [[#debugging][Debugging]]
-   - [[#help][Help]]
+ - [[Install][Install]]
```
Comment (LemmingAvalanche) [inline layers/+lang/ess/README.org]: Aren’t these links supposed to start with `#`? And if you change them to use `#` (and without initial capital letters) instead, you will get a smaller diff (reduced to your additions only).

I just ran `./core/tools/spacefmt/spacefmt all` while on your branch, which indeed changed these lines to `#`. So you should probably use `#`.

File: layers/+lang/ess/README.org [hunk 3]
```diff
@@ -3,57 +3,63 @@
...
+     - [[Inferior ESS Process Keybindings][Inferior ESS Process Keybindings]]
+     - [[ESS Watch][ESS Watch]]
+     - [[ESS Rdired][ESS Rdired]]
+     - [[ESS Data View][ESS Data View]]
 
 * Install
-To use this configuration layer, add it to your =~/.spacemacs=. You will need to
-add =ess= to the existing =dotspacemacs-configuration-layers= list in this
-file.
+  To use this configuration layer, add it to your =~/.spacemacs=. You will need to
```
Comment (LemmingAvalanche) [inline layers/+lang/ess/README.org]: Spurious change. Only changes whitespace (indentation). Same for many of the hunks below this one.

**Comment by LemmingAvalanche** [comment]
Btw: the email addresses of your “author” and “committer” are different on one of the commits, which makes one of the commits show up as “andrew committed with andrew” on GitHub. ☺

**Comment by nikolaiam** [comment]
R support was one of the chief reasons for me to start using Spacemacs, so very relevant to me as well! Would love to see this merged :blush:

**Comment by robbyoconnor** [comment]
@andrew-christianson https://travis-ci.org/syl20bnr/spacemacs/jobs/239746436

**Comment by TheBB** [comment]
I'd be happy to merge at least the first commit if you make the changes that have been proposed, and bring the company integration up to scratch with current Spacemacs.

**Pull request closed without merging**